### PR TITLE
Linting Improvement

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR reinforces the linting checks on Liqo code, by adding more controls and linters.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+run:
+  timeout: 10m
+  skip-files:
+    - "zz_generated.*.go"
+    - ".*_test.go"
```
Comment (giorio94) [inline .golangci.yml:5]: Woudn't skip test files

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  exclude-rules:
+    - linters:
+        - golint
+      # Excluding, as many packages are currently affected by this
+      text: "don't use an underscore in package name"
+
+    - linters:
+        - stylecheck
+      # Excluding, as many packages are currently affected by this
+      text: "ST1003: should not use underscores in package names"
```
Comment (giorio94) [inline .golangci.yml:115]: Are those exclude-rules required?

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    - goimports
+    - golint
+    - gomodguard
+  # - gomnd
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+  # - lll
```
Comment (giorio94) [inline .golangci.yml:77]: I would personally enable lll, to prevent excessively long lines (probably 120 is a good compromise)

**Review by giorio94 — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (palexster) [reply .golangci.yml:5]: I would skip tests in this first phase.

Comment (giorio94) [reply .golangci.yml:5]: Is it there any specific reason? Do you expect those to be characterized by more linting errors or is it just a matter of reducing the amount of files to be fixed?

**Pull request merged**